### PR TITLE
net/freeradius: options for fallback vlan

### PR DESCRIPTION
**Pull request by Alanin**
I have defined options fields in the general tab to be able to define a fallback group id and activate the option. also I have changed the config template for user-file generation to add the configured parameters when activated

### CODE REVIEW

**Comment by Alanin** [comment]
Hi, thank you very much for looking at the pullrequest. I have added the contraints. 

Your second point sounds valid. However, to be honest, I don't have enough knowledge to evaluate this concretely or to fix/work around the problem.

Would a hint in the GUI be enough here?

**Review by mimugmail — CHANGES_REQUESTED** [review]
Can you please bump the version in Makefile, and model, add a changelog to pkg-descr and also set "advanced" tags in forms so these two values are not shown. Also in help text add a warning that enabling this feature will sent an accepted radius reply also for false authentication.

**Comment by Alanin** [comment]
I am done...

simple options
image

advanced options
image

**Review by mimugmail — APPROVED** [review]
Thx :)

File: net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
         <label>Enable VLAN fallback assignment</label>
         <type>checkbox</type>
         <advanced>true</advanced>
-        <help>This allows you to define a fallback VLAN-Group-ID. Beware setting this option will send an accepted radius reply even the authentication fails.</help>
+        <help>This allows you to define a fallback VLAN-Group-ID. Be ware setting this option will send an accepted radius reply even the authentication fails.</help>
```
Comment (chelming) [inline net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml]: maybe "Be aware that setting this option will send an accepted RADIUS reply even if the authentication fails."

there's still a typo with "even the authentication"

File: net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml [hunk 2]
```diff
@@ -11,6 +11,20 @@
...
+    <field>
+        <id>general.fallbackvlan_enabled</id>
+        <label>Enable VLAN fallback assignment</label>
+        <type>checkbox</type>
+        <advanced>true</advanced>
+        <help>This allows you to define a fallback VLAN-Group-ID. Be aware setting this option will send an accepted radius reply even the authentication fails.</help>
+    </field>
+    <field>
+        <id>general.fallbackvlan_id</id>
+        <label>Fallback VLAN id</label>
```
Comment (chelming) [inline net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml:23]: `ID` should be capitalized

File: net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml [hunk 3]
```diff
@@ -11,6 +11,20 @@
...
+        <type>text</type>
+        <advanced>true</advanced>
+        <help>Define the Fallback VLAN group id.</help>
```
Comment (chelming) [inline net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml:26]: `ID` should be capitalized again

File: net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml (hunk 1 above)
Comment (chelming) [reply net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml]: also, `RADIUS` should be capitalized

Comment (chelming) [reply net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml]: perhaps, 
>Warning: Setting this option will send an accepted RADIUS reply even if the authentication attempt fails

Comment (Alanin) [reply net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml]: thank you. changed.

File: net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml (hunk 2 above)
Comment (Alanin) [reply net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml:23]: solved

**Comment by Alanin** [comment]
solving #2722 with this PR

**Comment by chelming** [comment]
I'm so excited for this. Thank you!

**Review by mimugmail — APPROVED** [review]
Still good to merge :)

File: net/freeradius/pkg-descr [hunk 4]
```diff
@@ -15,6 +15,9 @@ The server is fast, feature-rich, modular, and scalable.
 Plugin Changelog
 ================
 
+1.9.18
```
Comment (fichtner) [inline net/freeradius/pkg-descr:18]:
```suggestion
1.9.18

```

File: net/freeradius/src/opnsense/mvc/app/models/OPNsense/Freeradius/General.xml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <model>
     <mount>//OPNsense/freeradius/general</mount>
     <description>FreeRADIUS configuration</description>
-    <version>1.0.1</version>
+    <version>1.9.18</version>
```
Comment (fichtner) [inline net/freeradius/src/opnsense/mvc/app/models/OPNsense/Freeradius/General.xml:4]:
```suggestion
    <version>1.0.2</version>
```

File: net/freeradius/src/opnsense/service/templates/OPNsense/Freeradius/users [hunk 6]
```diff
@@ -98,4 +98,12 @@ DEFAULT Hint == "CSLIP"
 
 DEFAULT Hint == "SLIP"
         Framed-Protocol = SLIP
+
+{%  if helpers.exists('OPNsense.freeradius.general.fallbackvlan_enabled') and OPNsense.freeradius.general.fallbackvlan_enabled == '1' %}
```
Comment (fichtner) [inline net/freeradius/src/opnsense/service/templates/OPNsense/Freeradius/users:102]:
```suggestion
{%  if helpers.exists('OPNsense.freeradius.general.fallbackvlan_enabled') and OPNsense.freeradius.general.fallbackvlan_enabled == '1' %}

```

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**